qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -470,6 +470,28 @@ def valid_amp_conditions(self) -> sym.Expr:
         """Return symbolic expression for the pulse amplitude constraints."""
         return self._valid_amp_conditions
 
+    # This should be removed once the complex amp deprecation is completed.
+    @property
+    def amp(self) -> Union[ParameterExpression, complex]:
+        if "amp" in self._params:
+            if isinstance(self._params["amp"], complex):
+                warnings.warn(
+                    f"{self.__class__.__name__}.amp returns the complex amplitude which will be deprecated. "
+                    f"Use {self.__class__.__name__}.amp_angle to get the magnitude of the amplitude and the complex angle "
+                    "as a tuple",
+                    PendingDeprecationWarning
+                )
+            return self._params["amp"]
+        else:
+            raise AttributeError(f"'{self.__class__.__name__}' object has no attribute 'amp'")
+
+    @property
+    def amp_angle(self) -> tuple:
```

Comment:
User may want to check `.angle` property as well. Probably you can define subclass of `SymbolicPulse` with `self.amp` and `self.angle` (but need to be careful for round-trip test of QPY).

Also the user may expect angle in the `.parameters`. Because this is a restriction of the transport layer, to me updating the Pulse Qobj converter seems more reasonable.
https://github.com/Qiskit/qiskit-terra/blob/be1e6976138201cefa021e55dd2f0dac99bf6a1f/qiskit/qobj/converters/pulse_instruction.py#L427-L434
I like this approach because we can remove this ugly code.
https://github.com/Qiskit/qiskit-terra/blob/be1e6976138201cefa021e55dd2f0dac99bf6a1f/qiskit/pulse/library/symbolic_pulses.py#L432-L437
Basically `SymbolicPulse` itself doesn't have any restriction on parameter value type.

Reply:
Don't you think it will be confusing to have `pulse.amp` return the complex amplitude (for backward compatibility) while simultaneously having `pulse.angle`? I preferred to return the duo to avoid the confusion here. Again, I might be overthinking this.